### PR TITLE
HBCA-2 Search Component

### PR DESCRIPTION
**Pull request by heathharrelson**
Add a component that searches the Solr index for plain-language terms as the user types.

### CODE REVIEW

**Comment by heathharrelson** [comment]
@kirenotneb @wsams — I'm happy to walk through the code with you if that would save time.

File: src/components/SearchForm.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    <div class="dropdown-pane" :class="{'is-open': hasSuggestions }" data-dropdown>
+      <ul>
+        <li v-for="(suggestion, index) in suggestions"
+            :key="suggestion.id"
+            :class="{'highlighted': isActive(index)}"
+            @click="suggestionClick(index)"
+            @mouseover="mouseOver(index)"
+            @mouseout="mouseOut()"
+        >
+          <strong>{{ suggestion.exact_synonym[0] }}</strong><br/>
```
Comment (wsams) [inline src/components/SearchForm.vue]: Is the space after `{{` and before `}}` also a default style?

Comment (heathharrelson) [reply src/components/SearchForm.vue]: This is style I copied from the component this is based on ([see here](https://github.com/putmantime/vue_autocomplete/blob/master/src/components/AutoComplete.vue)), but I don't feel strongly about it. ESLint isn't complaining if I remove the spaces, so I'm happy to do that if it's more readable.

Comment (heathharrelson) [reply src/components/SearchForm.vue]: Oh, and it's used in the Vue documentation as well: https://vuejs.org/v2/guide/#Declarative-Rendering

**Comment by wsams** [comment]
It's readable either way, was just curious.

**Pull request merged**